### PR TITLE
Update plugins/fields/media/media.php to prevent deprecated message caused if $value empty.

### PR DESCRIPTION
**Pull request by particthistle**
### Summary of Changes
PHP error was encountered when test instructions below are met:

`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

If $value is null at line 86 in plugins/fields/media/media.php, then the deprecated error occurs.

Added if statement to checkValue function to check $value is set before trying to decode the string.

### Testing Instructions

- Create a custom field for articles that is a Media type, that's also set to optional (ie Required = No)
- Create an article, leaving the optional custom fields blank.
- View the article in the front end.
- Turn on Error Reporting for the site to Development so you can see the messages on screen.

(ShackOpenGraph also installs Social Images fields that are Media type and trigger the error)

### Actual result BEFORE applying this Pull Request

If there's no image in the custom field, then the code in the plugin does a check on the $value provided to the checkValue function. As there's no value, json_decode triggers 
`json_decode(): Passing null to parameter #1 ($json) of type string is deprecated`

### Expected result AFTER applying this Pull Request
The checkValue function will now firstly check whether $value is null, and then only proceed with the rest of the function when $value has a value provided to it.

This in turn stops the deprecated message being generated.

### Link to documentations
Please select:
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/fields/media/media.php [hunk 1]
```diff
@@ -83,13 +83,14 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
      */
     private function checkValue($value)
     {
-        json_decode($value);
+        if ($value) {
```
Comment (laoneo) [inline plugins/fields/media/media.php]: If you invert the check, then the whole functuon becomes easier to read. Like if (! $value) {return ;}

Comment (particthistle) [reply plugins/fields/media/media.php]: Thanks @laoneo - makes sense... please check change I made makes sense.

**Review by laoneo — APPROVED** [review]

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 9e169dff65f82ce4406bf8c85a5167510c991af7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39542">issues.joomla.org/tracker/joomla-cms/39542</a>.</sub>

**Comment by particthistle** [comment]
With 4.2-dev locked for 4.3 release, rebased this PR to 4.3-dev.

**Comment by joomdonation** [comment]
@particthistle Actually, 4.2-dev still accepts bugs fix. So you should not have to rebase to 4.3-dev.

**Comment by obuisard** [comment]
It looks like something went wrong with the files associated with the PR. Can you fix it on your end Patrick @particthistle? Thanks, I appreciate it.

**Comment by particthistle** [comment]
Was nudged on this during PBF. Aside from botching the rebasing of the PR, and being away 3 months due to finding a new house and moving, the problem preventing the code being merged is more due to the entire Custom Fields code being refactored for Joomla 4.3 from what I have worked out. Had it been actioned back in February it would not have had an issue.

https://github.com/joomla/joomla-cms/blob/4.4-dev/plugins/fields/media/src/Extension/Media.php shows where the code has moved to.

In trying to replicate the problem with 4.3.3 and 4.3.4.rc1 it appears to have been resolved in the refactoring process.

If it does turn out that it needs to be fixed (scenario may be where data has been imported from J3 to J4, and the initial values for the data is NULL instead of a correct default blank value) then the code to add into a new PR before [Line 89](https://github.com/joomla/joomla-cms/blob/d20c4557cea681d02aeceee97e9ece795961da2f/plugins/fields/media/src/Extension/Media.php#L89) is:

`        // If the value is null return as json_decode will cause a deprecated issue with json_decode
        if (!$value)
        {
            return;
        }`

However as I can't replicate the problem with the above testing instructions, I'm closing this PR at the moment.

If anyone wants to walk me through the issue, or help with creating some "Github PR for Joomlers" content, reach out on Mattermost.

**Pull request closed without merging**